src/middleware/error-handler.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+
+// Define a custom error type if needed
+interface CustomError extends Error {
+  statusCode?: number;
+  code?: number;
+  errors?: Record<string, { message: string }>;
+  keyValue?: Record<string, any>;
+  value?: string;
+}
+
+export const errorHandlerMiddleware = (
+  err: CustomError, // Ensure this is typed as a CustomError
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void => {
+  let customError = {
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || "Something went wrong, try again later",
+  };
+
+  if (err.name === "ValidationError") {
```

Comment:
Where are these three error types produced? This is helpful context that should be included in a comment or the commit message. If we don't produce these error types yet, let's leave this until a later commit where we do.